### PR TITLE
fix: add quickstart in demo and fix docs

### LINKED ISSUE #2824: macos java sdk doesn't support macOS 11 & 10.15
macos-latest is 12 now, some users found that macos java sdk can't load the shared library successfully in macOS 11.
And the native jar built by macOS 11 works.
If anyone needs macOS 11 support, trigger the ci on the patch https://github.com/vagetablechicken/fedb/commit/1d7a38e410254c5b07634f914494e12e1674f585, notice that ./java/prepare_release.sh "0.7.3" is set manually.

for macOS 10.15, use macos sdk 10.15(xcode switch) to build on macOS 11, but haven't been tested, ref https://github.com/vagetablechicken/fedb/blob/069-mac11-build10.15/.github/workflows/sdk.yml

### PR DESCRIPTION
**Pull request by vagetablechicken**
fix quickstart docs and close #2824

quickstart demo can be the individual repo in demo, so we can use docker-compose to test later.

### CODE REVIEW

File: demo/java_quickstart/demo/src/main/java/com/openmldb/demo/App.java [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+            demo.insertWithoutPlaceholder();
+            // 通过placeholder的方式插入。placeholder方式不会重复编译sql, 在性能上会比直接insert好很多
+            demo.insertWithPlaceholder();
+            // 执行select语句
+            demo.select();
+            // 在request模式下执行sql
+            demo.requestSelect();
+            // 删除表
+            demo.dropTable();
+          	// 删除数据库
```
Comment (dl239) [inline demo/java_quickstart/demo/src/main/java/com/openmldb/demo/App.java]: rm redundant blank

File: demo/java_quickstart/demo/pom.xml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.11</version>
+      <scope>test</scope>
+    </dependency>
+
```
Comment (dl239) [inline demo/java_quickstart/demo/pom.xml:28]: if run on macOS, `openmldb-native` should be specified

File: docs/zh/quickstart/sdk/python_sdk.md [hunk 3]
```diff
@@ -208,7 +209,11 @@ openmldb.sql_magic.register(db)
 
 ![](./images/openmldb_magic_function.png)
 
-## 5. Option详解
+## 完整使用范例
+
+见[Python quickstart demo](https://github.com/4paradigm/OpenMLDB/tree/main/demo/python_quickstart/demo.py)，包括了上文的DBAPI和SQLAlchemy用法。
```
Comment (dl239) [inline docs/zh/quickstart/sdk/python_sdk.md:214]: update to relative path?

Comment (vagetablechicken) [reply docs/zh/quickstart/sdk/python_sdk.md:214]: documents is a individual website, not in the github repo

File: demo/java_quickstart/demo/pom.xml (hunk 2 above)
Comment (vagetablechicken) [reply demo/java_quickstart/demo/pom.xml:28]: I'll add a comment in quickstart doc.

**Review by zhanghaohit — APPROVED** [review]

File: docs/zh/deploy/compile.md [hunk 4]
```diff
@@ -51,7 +51,7 @@ make NPROC=4
 ### 依赖工具
 
 - gcc >= 8 或者 AppleClang >= 12.0.0
-- cmake 3.20 或更新版本
+- cmake 3.20 或更新版本（建议 < cmake 3.24）
```
Comment (aceforeverd) [inline docs/zh/deploy/compile.md:54]: related to #2883,  doc need update after the issue resolved.

File: demo/java_quickstart/demo/pom.xml [hunk 5]
```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
```
Comment (aceforeverd) [inline demo/java_quickstart/demo/pom.xml:2]: does docker-compose test covered ?

Comment (vagetablechicken) [reply demo/java_quickstart/demo/pom.xml:2]: Next pr, it's a big change, we'll add 3 quickstart demo and some use cases.

**Review by aceforeverd — APPROVED** [review]

**Pull request merged**